### ISSUE TITLE
FR: Allow setting default filetype (or at least default file extension) for new Geany tabs

### ISSUE THREAD

**Issue by Gitoffthelawn** [geany/geany#3847]
After opening Geany, or after creating a new tab in Geany, the default filetype is `None` and the default extension is null.

It will be quite useful to allow the user to specify a default filetype, or at least a default file extension when saving content in new Geany tabs.

A possible simple implementation would be allow the user to create `filetypes.None.conf` in which they could specify something like:

```
[settings]
# default extension used when saving files
extension=txt
```

**Comment by elextr** [comment]
Having the ability to set the filetype of new files is reasonable, just add a setting to `Edit->Preferences->Files->New Files` with a combo box to select the filetype.  The semantics can simply be the same as `Files->New` then `Document->Set filetype->[what the setting says]`.

**Comment by Gitoffthelawn** [comment]
> Having the ability to set the filetype of new files is reasonable, just add a setting to `Edit->Preferences->Files->New Files` with a combo box to select the filetype. The semantics can simply be the same as `Files->New` then `Document->Set filetype->[what the setting says]`.

Yes, that will work quite well.

**Comment by blimeybloke** [comment]
Following this as it is a feature I would love to see.

**Comment by rickyrockrat** [comment]
Yes. I do a lot of bash scripting with include files with no extension. I need to set the 'none' extension to shell. I tried modifying the existing filetypes.sh and set the extension to null, but that didn't work. However, when I added this line to filetypes.conf (note the ending *):
Sh=APKBUILD;*.sh;configure;configure.in;configure.in.in;configure.ac;*.ksh;*.mksh;*.zsh;*.ash;*.bash;.bashrc;bash.bashrc;.bash_*;bash_*;*.m4;PKGBUILD;*profile;*

Files without extensions now show highlight as shell scripts! Sweet! This is on Pryce (built on or after 2024-05-20)
